addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1751,7 +1751,7 @@ var SnippetsMenu = Widget.extend({
      * @returns {Promise<SnippetEditor>}
      *          (might be async when an editor must be created)
      */
-    _activateSnippet: async function ($snippet, previewMode, ifInactiveOptions) {
+    _activateSnippet: async function ($snippet, previewMode, ifInactiveOptions, inMutex) {
```

Comment:
@qsm-odoo `_activateSnippet` tries to take the lock when it is already taken by one of the method higher in the call stack (`_onSnippetEditionRequest`). Since we await on `_activateSnippet` we create a deadlock... Not sure how to fix this. The current code is just here to show from where comes the problem.